src/full/Agda/TypeChecking/Reduce.hs

```diff
@@ -413,14 +413,21 @@ instance Reduce Sort where
             -- don't instantiate we might end up blocking on a solved
             -- metavariable.
             s2' <- instantiateFull s2'
+            -- piSort' should only be called on reduced sorts
+            s1' <- reduce s1'
+            s2' <- reduce s2'
```

Comment:
The sorts s1' and s2' are already reduced by the call to `reduceB` above, so there should be no need to reduce them again.